news/_posts/2020-12-09-pr-testing-on-prestashop.md

```diff
@@ -0,0 +1,179 @@
+
+![A comment about a bug on a PR](/assets/images/2020/12/qa_return_pr.png)
+
+### And if everything is OK?
+
+If we don't find any bug during our test, then we change the label to “QA Approved”. This allows the maintainers to finally merge the PR in the branch. The change will be available on PrestaShop in the next build of the branch!
+
+![QA approved!](/assets/images/2020/12/qa_approved.png)
+
+Note that everything the QA specialists test is tracked on an internal tool (we use Testrail): it allows us, for example, to resume a test started by another member of the QA team, or to have the detail of what was checked on a specific PR.
+
+# III. Improvement Pull Request
+
+The process for testing a Pull Request that adds an improvement to the solution is quite similar to that of a bug fix Pull Request.
+
+### Requirements
+
+The requirements are almost the same for a bug fix. The main difference is that the new feature should be adequately [specified in writing](https://github.com/PrestaShop/prestashop-specs/) and greenlighted by the Product team or the maintainers: not all improvements are necessarily accepted, so it is important to discuss them with the team in an issue before submitting a PR.
+
+The “How to test” field should describe the new behavior, the nominal test cases, possible errors and how they should be handled, etc... unless formal acceptance tests have been included in the specification.
+Written specifications should be linked in the PR as well.
+
+If some information is missing, the QA analyst will ping the author, the product team and/or the maintainers and wait for this information to be completed before starting to work on the PR.
+
+### Check that the improvement works accordingly to the issue and specification
+
+This step is simple: using the information in the issue, the "How to test" instructions, and the specification (if there is one), then QA analyst proceeds to check out the PR and try out each new behavior to make sure it works as expected and described.
+
+### Regression testing
+
+For improvements, the regression test part is even more important than for bug fixes: the QA analyst must make sure that the original features the PR improves are still working as expected.
+
+To do so, we generally compare two installs (one on the branch with the PR and one without), then perform the same actions on each install, and verify that the result is strictly identical. We can also carry out parts of the test campaign usually performed to verify build releases.
+
+### What happens next?
+
+Like with bug fixes, if the QA analyst finds any bug, they will comment the PR with all useful tips to help the author fix the bug.
+If the QA analyst notices an unspecified behavior (if the expected behavior is unclear, or for example, if it could be argued that an error should be displayed in a given scenario) they will ping the Product team and/or the maintainers to discuss this particular issue.
```

Comment:
```suggestion
If the QA analyst notices an unspecified behavior (if the expected behavior is unclear, or for example, if it could be argued that an error should be displayed in a given scenario) they will ping the product team and/or the maintainers to discuss this particular issue.
```